cmd/kube-proxy/app/server.go

```diff
@@ -751,14 +750,14 @@ func (s *ProxyServer) Run() error {
 	serviceConfig.RegisterEventHandler(s.Proxier)
 	go serviceConfig.Run(wait.NeverStop)
 
-	if s.UseEndpointSlices {
+	if endpointsProxier, ok := s.Proxier.(config.EndpointsHandler); ok {
+		endpointsConfig := config.NewEndpointsConfig(informerFactory.Core().V1().Endpoints(), s.ConfigSyncPeriod)
+		endpointsConfig.RegisterEventHandler(endpointsProxier)
+		go endpointsConfig.Run(wait.NeverStop)
+	} else {
 		endpointSliceConfig := config.NewEndpointSliceConfig(informerFactory.Discovery().V1().EndpointSlices(), s.ConfigSyncPeriod)
 		endpointSliceConfig.RegisterEventHandler(s.Proxier)
 		go endpointSliceConfig.Run(wait.NeverStop)
-	} else {
-		endpointsConfig := config.NewEndpointsConfig(informerFactory.Core().V1().Endpoints(), s.ConfigSyncPeriod)
-		endpointsConfig.RegisterEventHandler(s.Proxier)
-		go endpointsConfig.Run(wait.NeverStop)
 	}
 
 	// This has to start after the calls to NewServiceConfig and NewEndpointsConfig because those
```

Comment:
`	if utilfeature.DefaultFeatureGate.Enabled(features.TopologyAwareHints) {`
requires slices IIRC

Reply:
It requires slices to actually be _useful_, but it appears that the code will happily do nothing if you try to enable it when using Endpoints.